supafunc/_async/functions_client.py

```diff
@@ -0,0 +1,83 @@
+from httpx import HTTPError, Response
+from typing import Any, Dict, Literal, Optional, Union
+
+from ..errors import FunctionsHttpError, FunctionsRelayError
+
+from ..utils import AsyncClient, __version__
+
+
+class AsyncFunctionsClient:
+    def __init__(self, url: str, headers: Dict):
+        self.url = url
+        self.headers = {
+            "User-Agent": f"supabase-py/storage3 v{__version__}",
+            **headers,
+        }
+
+    async def _request(
+        self,
+        method: Literal["GET", "OPTIONS", "HEAD", "POST", "PUT", "PATCH", "DELETE"],
+        url: str,
+        headers: Union[Dict[str, str], None] = None,
+        json: Optional[dict[Any, Any]] = None,
+    ) -> Response:
+        async with AsyncClient(base_url=self.url, headers=self.headers) as client:
```

Comment:
I think creating the AsyncClient instance in the init is causing the connection to not close as I've been seeing user's report.

Reply:
I think that one is because they're calling some sign_in_x method that starts a thread which only closes when they call a sign_out? I'm not very familiar with the gotrue-py code but I remember seeing an issue like that